pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/instance/PhysicianInstitutionComponent.java

```diff
@@ -20,56 +25,56 @@
-                                            String subtitleText,
-                                            InstitutionType institutionType,
-                                            boolean showFields,
-                                            boolean required,
-                                            boolean hideNumber) {
+    protected PhysicianInstitutionComponent(
+            ComponentType physicianOrInstitution,
+            InstitutionPhysicianComponentDto instDto,
+            boolean shouldHideNumber
+    ) {
         super(physicianOrInstitution);
         if (physicianOrInstitution != ComponentType.PHYSICIAN && physicianOrInstitution != ComponentType.INSTITUTION) {
             throw new IllegalArgumentException("Physician/Institution component must be either " + ComponentType
                     .PHYSICIAN + " or " + ComponentType.INSTITUTION);
         }
-        this.allowMultiple = allowMultiple;
-        this.addButtonText = addButtonText;
-        this.titleText = titleText;
-        this.subtitleText = subtitleText;
-        this.institutionType = institutionType;
-        this.showFields = showFields;
-        this.hideDisplayNumber = hideNumber;
-        this.required = required;
-        initParametersMap();
+        this.instDto = instDto;
+        this.hideDisplayNumber = shouldHideNumber;
     }
 
     private final void initParametersMap() {
-        parameters.put(ALLOW_MULTIPLE, allowMultiple);
-        parameters.put(ADD_BUTTON_TEXT, addButtonText);
+        parameters.put(ALLOW_MULTIPLE, instDto.getAllowMultiple());
+        parameters.put(ADD_BUTTON_TEXT, buttonText);
         parameters.put(TITLE_TEXT, titleText);
         parameters.put(SUBTITLE_TEXT, subtitleText);
-        parameters.put(INSTITUTION_TYPE, institutionType.name());
-        parameters.put(SHOW_FIELDS, showFields);
-        parameters.put(REQUIRED, required);
+        parameters.put(INSTITUTION_TYPE, instDto.getInstitutionType().name());
+        parameters.put(SHOW_FIELDS, instDto.showFields());
+        parameters.put(REQUIRED, instDto.isRequired());
+    }
+
+    @Override
+    public void registerTemplateIds(Consumer<Long> registry) {
+        super.registerTemplateIds(registry);
+        Optional.ofNullable(instDto.getButtonTemplateId()).ifPresentOrElse(registry::accept, () -> { });
+        Optional.ofNullable(instDto.getTitleTemplateId()).ifPresentOrElse(registry::accept, () -> { });
+        Optional.ofNullable(instDto.getSubtitleTemplateId()).ifPresentOrElse(registry::accept, () -> { });
+    }
+
+    @Override
+    public void applyRenderedTemplates(Provider<String> rendered, ContentStyle style) {
+        super.applyRenderedTemplates(rendered, style);
+        buttonText = rendered.get(instDto.getButtonTemplateId());
+        titleText = rendered.get(instDto.getTitleTemplateId());
+        subtitleText = rendered.get(instDto.getSubtitleTemplateId());
+
+        initParametersMap();
     }
```

Comment:
Not sure what to do with `style` here.

Reply:
Rendering is called from `FormInstance` and it traverses all the sections and blocks. We need to update `ComponentBlock` to call `FormComponent`'s rendering methods.

For the `style`, if it is `basic` we should strip the various strings into plain text. I think most of these attributes for components are text instead of HTML, so not sure if we should do it. But maybe we should do it for consistency and just-in-case? See other implementations of this method for how it's done (usually calling `HtmlConverter.getPlainText()`).